pkg/kubernetes/deploy.js

```diff
@@ -0,0 +1,454 @@
+                        rcs.push(jdata);
+                    } else if (jdata.kind === NS) {
+                        namespaces.push(jdata);
+                    } else {
+                        text = _("Unsupported Entity. ");
+                        file_note.show().text(text);
+                        return;
+                    }
+                }
+            } else {
+                text = _("Unable to Read the file.Please check the json file. ");
+                file_note.show().text(text);
+                return;
+            }
+
+            create_everything(namespace, services, rcs, pods)
+                .progress(function(code, response, n, total) {
+                    //TODO Progress bar
+
+                })
+                .done(function() {
+                    /* code gets run when everything is created */
+                    disable_deploy_button();
+                    hide_progress_message();
+                     $('#deploy-app-dialog').modal('hide');
+
+                })
+                .fail(function(ex, response) {
+                    /* ex containst the failure */
+                    enable_deploy_button();
+                    var jdata = JSON.parse(response);
+                    hide_progress_message();
+                    $("#deploy-app-general-error-msg").text(jdata.message).parent().show();
+                    console.warn(jdata.message);
+                });
+        }
+
+        function create_everything(namespace, services, rcs, pods) {
+            var ns_json = {"apiVersion":"v1beta3","kind":"Namespace","metadata":{"name": "" }};
+            ns_json.metadata.name = namespace;
+            var tasks = [];
+
+            tasks.push([namespace, [JSON.stringify(ns_json)], client.create_ns]);
+
+            for (var serv in services) {
+                tasks.push([services[serv].metadata.name, [namespace, JSON.stringify(services[serv])], client.create_service]);
+            }
+
+            for (var rc in rcs) {
+                tasks.push([rcs[rc].metadata.name, [namespace, JSON.stringify(rcs[rc])], client.create_replicationcontroller]);
+            }
+
+            for (var p in pods) {
+                tasks.push([pods[p].metadata.name, [namespace, JSON.stringify(pods[p])], client.create_pod]);
+            }
+
+            var deferred = $.Deferred();
+            var total = tasks.length;
+
+            function step() {
+                var e = tasks.shift();
+                if (!e) {
+                    deferred.resolve();
+                    return;
+                }
+                var task_name = e[0];
+                var args = e[1];
+                var task = e[2];
+
+                task.apply(null, args).done(function(response) {
+                    deferred.notify("created", response, tasks.length - total, total);
+                    step();
+                }).fail(function(ex, response) {
+                    if (ex.status == 409) {
```

Comment:
This does not match the design. The design shows that if an object exists, the operation fails. Do you have an explanation somewhere of why this approach is better? Mentioned this previously here:

https://github.com/cockpit-project/cockpit/pull/1989#issuecomment-93670018